docs/docs/glossary.md

```diff
@@ -2,7 +2,240 @@
+### Filesystem
+
+The way files are organized. But with Gatsby it means having files in the same place as your website's or app's code instead of having data from an external [source](#source)
+
+### Front-End
+
+The [public-facing](#public) interface for your website or app.
+
+## G
+
+### Gatsby
+
+This project. A framework based on [React](#react) that helps you build blazing fast websites and apps.
+
+### GraphQL
+
+A [query](#query) language that allows you to pull data into your website or app. It’s the interface that Gatsby uses for managing site data.
+
+## H
+
+### Hosting
+
+A hosting provider keeps a copy of your website or app and makes it accessible to [the public](#public).
+
+### HTML
+
+A markup language that every web browser is able to understand. It stands for HyperText Markup Language. HTML gives your web content a universal informational structure, defining things like headings, paragraphs, and more.
+
+### Headless CMS
+
+A [CMS](#cms) that only handles the [back-end](#back-end) content management instead of handling both the back-end and [front-end](#frontend).
+
+## I
+
+## J
+
+### JavaScript
+
+A programming language that helps us make the web dynamic and interactive.
+
+## K
+
+## L
+
+## M
+
+### Markdown
+
+A way of writing content in plain text, using special characters such as underscores and asterisks to format content.
+
+### MDX
+
+Extends [Markdown](#markdown) to support [React](#react) [components](#components) within your content.
+
+## N
+
+### Node
+
+This might refer to Node.js - a program that lets you run [JavaScript](#javascript) on your computer. Gatsby is powered by Node.
+
+In Gatsby it may also refer to a data node. A single piece of data. A [data source](#data-source) will create multiple nodes.
+
+### NPM
+
+[Node](#node) Package Manager. Allows you to install and update other packages that your project depends on. [Gatsby](#gatsby) and [React](#react) are examples of your project's dependencies. See also: [Yarn](#yarn).
+
+## O
+
+## P
+
+### Page
+
+An [HTML](#html) page.
+
+This also often refers to [components](#components) that live in `/src/pages/` and are converted to pages by Gatsby.
+
+### Plugin
+
+Additional code that adds functionality to Gatsby that wasn't included out-of-the-box.
+
+### Public
+
+This usually refers to either a member of the public (as opposed to your team) or the folder `/public` in which your [built](#build) website or app is saved.
+
+### Programmatically
+
+Something that automatically happens based on your code and configuration. For example, you might [configure](#config) your project to create a [page](#page) for every blog post written.
+
+### Production Environment
+
+The [environment](#environment) for the [built](#build) website or app. It's accessed through the [CLI](#cli) using `gatsby build` or `gatsby serve`.
+
+## Q
+
+### Query
+
+The process of requesting specific data from somewhere. With Gatsby you normally query with [GraphQL](#graphql).
+
+## R
```

Comment:
We could add "Router" here to talk a bit about the client-rendering part of Gatsby.  https://www.gatsbyjs.org/docs/routing/

Reply:
Could you give me a hand with this definition please? Struggling to come up with something that doesn't sound strange and I'm not sure my definition is actually correct.

```
Routing allows [Pages](#pages) and [Components](#components) to be accessible via a URL.
```

Reply:
Sure thing! I wrote about routing in a [blog post](https://marcysutton.com/links-vs-buttons-in-modern-web-applications) once...how about something like this?
> Much like filtering in a mail room, routing is the mechanism for connecting network requests with the appropriate content in a web application. Routers like `@reach/router`–used in Gatsby–are common in JavaScript web frameworks: they map URL fragments with views and subviews, enabling dynamic URLs that don't have to be written by hand.

Reply:
Cool, thanks. I submitted a tweaked version but strangely I had to do a new pull request for it (I'm probably misusing GitHub) https://github.com/gatsbyjs/gatsby/pull/13739

Reply:
It wasn't you, there was a configuration change that needed some adjustment. It should be resolved soon if it hasn't already!